wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java

```diff
@@ -1077,13 +1078,18 @@ public int randomGet(Memory memory, int buf, int bufLen) {
         if (bufLen < 0) {
             return wasiResult(WasiErrno.EINVAL);
         }
-        if (bufLen >= 100_000) {
-            throw new WASMRuntimeException("random_get: bufLen must be < 100_000");
-        }
 
-        byte[] data = new byte[bufLen];
-        new SecureRandom().nextBytes(data);
-        memory.write(buf, data);
+        byte[] data = new byte[min(bufLen, 4096)];
+        int written = 0;
+        while (written < bufLen) {
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
This is interesting, basically we should have this check in any "long enough" operation.

Reply:
Yeah, I was thinking we might want an `ExecutionInterruptedException` or `ExecutionAbortedException` with a static utility method to do this check

Reply:
:+1: